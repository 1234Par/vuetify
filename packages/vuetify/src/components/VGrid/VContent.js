// Styles
import '../../stylus/components/_content.styl'

// Mixins
import SSRBootable from '../../mixins/ssr-bootable'

/* @vue/component */
export default {
  name: 'v-content',

  mixins: [SSRBootable],

  props: {
    tag: {
      type: String,
      default: 'main'
    }
  },

  computed: {
    styles () {
      const {
        bar, top, right, footer, insetFooter, bottom, left
      } = this.$vuetify.application

      const offset = this.$vuetify.breakpoint.mdAndUp ? 24 : 16

      return {
        paddingTop: `${top + bar + offset}px`,
        paddingRight: `${right + offset}px`,
        paddingBottom: `${footer + insetFooter + bottom + offset}px`,
        paddingLeft: `${left + offset}px`
      }
    }
  },

  render (h) {
    const data = {
      staticClass: 'v-content',
      style: this.styles,
      ref: 'content'
    }

    return h(this.tag, data, [
      h(
        'div',
        { staticClass: 'v-content__wrap' },
        this.$slots.default
      )
    ])
  }
}
