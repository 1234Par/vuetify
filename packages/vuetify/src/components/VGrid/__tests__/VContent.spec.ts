// Components
import VContent from '../VContent'

// Services
import { Application } from '../../../services/application'

// Utilities
import {
  mount,
  MountOptions,
  Wrapper
} from '@vue/test-utils'
import { rafPolyfill } from '../../../../test'

describe('VContent.ts', () => {
  type Instance = InstanceType<typeof VContent>
  let mountFunction: (options?: MountOptions<Instance>) => Wrapper<Instance>

  beforeEach(() => {
    mountFunction = (options = {} as MountOptions<Instance>) => {
      return mount(VContent, {
        mocks: {
          $vuetify: {
            application: new Application()
          }
        },
        ...options
      })
    }
  })

  rafPolyfill(window)

  it('should render', () => {
    const wrapper = mountFunction()

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render with custom tag', () => {
    const wrapper = mountFunction({
      propsData: {
        tag: 'div'
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should calculate paddings', () => {
    const wrapper = mountFunction({
      mocks: {
        $vuetify: {
          application: {
            bar: 1,
            top: 2,
            right: 3,
            footer: 4,
            insetFooter: 5,
            bottom: 6,
            left: 7
          }
        }
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
