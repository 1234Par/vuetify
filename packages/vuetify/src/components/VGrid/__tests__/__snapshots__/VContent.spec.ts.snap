// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VContent.ts should calculate paddings 1`] = `

<main class="v-content"
      style="padding: 3px 3px 15px 7px;"
>
  <div class="v-content__wrap">
  </div>
</main>

`;

exports[`VContent.ts should render 1`] = `

<main class="v-content"
      style="padding: 0px 0px 0px 0px;"
>
  <div class="v-content__wrap">
  </div>
</main>

`;

exports[`VContent.ts should render with custom tag 1`] = `

<div class="v-content"
     style="padding: 0px 0px 0px 0px;"
>
  <div class="v-content__wrap">
  </div>
</div>

`;
